code/game/objects/effects/temporary_visuals/temporary_visual.dm

```diff
@@ -34,3 +34,43 @@
 	if(set_dir)
 		setDir(set_dir)
 	return ..()
+
+
+///Image that appears at the Xeno Rally target; only Xenos can see it
+/obj/effect/temp_visual/xenomorph/xeno_tracker_target
+	name = "xeno tracker target"
+	icon_state = "nothing"
+	duration = XENO_HEALTH_ALERT_POINTER_DURATION
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	color = COLOR_RED
+	hud_possible = list(XENO_TACTICAL_HUD)
+	///The target we're pinging and adding this effect to
+	var/atom/tracker_target
+	///The visual effect we're attaching
+	var/image/holder
+
+/obj/effect/temp_visual/xenomorph/xeno_tracker_target/Destroy()
+	. = ..()
+	if(tracker_target && holder) //Check to avoid runtimes
+		tracker_target.overlays -= holder //remove the overlay
+	tracker_target = null //null the target var
+	QDEL_NULL(holder) //remove the holder
+	holder = null //null the holder var
```

Comment:
```suggestion
	QDEL_NULL(holder) //remove the holder and null the var
```

Reply:
I'm fairly certain this doesn't work, but I'll do it to get the merge